### ISSUE TITLE
RGCNConv update function fails when root_weight=False

### ISSUE THREAD

**Issue by arthurfeeney** [rusty1s/pytorch_geometric#961]
## 🐛 Bug

I might have stumbled on a bug. (If not, apologies! I'm new to this.) I didn't find another issue posted about it.
When using RGCNConv with root_weight=False, the update function fails. When root_weight is False, the "out" variable is never defined in update. So, depending on whether bias is True or False, the function either fails to add the bias or has no "out" variable that it can return.

## To Reproduce

1. Just make a basic network that has a layer like this: RGCNConv(in_dim, hidden_dim, num_relations=nr, num_bases=nb, root_weight=False)
2. Train it normally!

The network that I find this with is for multi-type link prediction (and very simple). It works fine when root_weight = True.

I was able to reproduce this by modifying the "Learning Methods on Graphs" section of "Introduction by Example" in the docs, so that it used RGCNConv with num_relations=1 (and passed a tensor of zeros for edge_type in forward). Again, this worked fine when root_weight=True and failed when root_weight=False. Here is the network that failed:

  8 class Net(torch.nn.Module):                                                    
  9     def __init__(self, device):                                                
 10     ┆   super(Net, self).__init__()                                            
 11     ┆   self.conv1 = RGCNConv(dataset.num_node_features, 16, num_relations=1,  
 12     ┆   ┆   ┆   ┆   ┆   ┆   ┆ num_bases=1, root_weight=True)                   
 13     ┆   self.conv2 = RGCNConv(16, dataset.num_classes, num_relations=1,        
 14     ┆   ┆   ┆   ┆   ┆   ┆   ┆ num_bases=1, root_weight=False)                  
 15     ┆   self.device = device                                                   
 16                                                                                
 17     def forward(self, data):                                                   
 18     ┆   x, edge_index = data.x, data.edge_index                                
 19     ┆   edge_type = torch.zeros(edge_index.size(1)).to(self.device).long()     
 20     ┆   x = self.conv1(x, edge_index, edge_type)                               
 21     ┆   x = F.relu(x)                                                          
 22     ┆   x = F.dropout(x, training=self.training)                               
 23         x = self.conv2(x, edge_index, edge_type)                               
 24     ┆   return F.log_softmax(x, dim=1)   

Here is the error message when root_weight=False and bias=True. A similar error occurs when both root_weight and bias are False.

Traceback (most recent call last):
  File "./src/main.py", line 181, in <module>
    main()
  File "./src/main.py", line 140, in main
    loss += train(batch, model, optimizer)
  File "./src/main.py", line 70, in train
    pred = model(data.x, data.edge_index, data.edge_attr)
  File "/mnt/nfs/scratch1/afeeney/miniconda3/envs/ddime/lib/python3.8/site-packages/torch/nn/modules/module.py", line 532, in __call__
    result = self.forward(*input, **kwargs)
  File "/mnt/nfs/work1/696ds-s20/afeeney/ddime/src/models/basic_gnn.py", line 32, in forward
    x = self.conv1(x, edge_index=edge_index,
  File "/mnt/nfs/scratch1/afeeney/miniconda3/envs/ddime/lib/python3.8/site-packages/torch/nn/modules/module.py", line 532, in __call__
    result = self.forward(*input, **kwargs)
  File "/mnt/nfs/scratch1/afeeney/miniconda3/envs/ddime/lib/python3.8/site-packages/torch_geometric/nn/conv/rgcn_conv.py", line 70, in forward
    return self.propagate(edge_index, size=size, x=x, edge_type=edge_type,
  File "/mnt/nfs/scratch1/afeeney/miniconda3/envs/ddime/lib/python3.8/site-packages/torch_geometric/nn/conv/message_passing.py", line 179, in propagate
    out = self.update(out, **update_kwargs)
  File "/mnt/nfs/scratch1/afeeney/miniconda3/envs/ddime/lib/python3.8/site-packages/torch_geometric/nn/conv/rgcn_conv.py", line 97, in update
    out = out + self.bias
UnboundLocalError: local variable 'out' referenced before assignment
  

## Expected behavior

To put it simply, I was expecting my model to continue working, but not use a root weight!
If root_weight is False, it seems like the output of the update function should just be something like "bias + aggr_out".  From looking at some other modules, maybe it should be something like the update function in SplineConv and NNConv? I.e., the update function can accumulate onto aggr_out, rather than define a new "out" variable?

## Environment

* OS: CentOS 7
* Python version: 3.8.1
* PyTorch version: 1.4.0
* CUDA/cuDNN version: CUDA is 10.1 / cuDNN is 7.6 
* GCC version: 5.4.0

**Comment by rusty1s** [comment]
This is indeed a bug, thank you very much! I fixed it in master.

**Comment by douglasrizzo** [comment]
Maybe this should be closed, as I believe the fix is already present in the latest release.
